src/app/Rules/Rules.tsx

```diff
@@ -0,0 +1,197 @@
+  const addSubscription = useSubscriptions();
+
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [sortBy, setSortBy] = React.useState({} as ISortBy);
+  const [rules, setRules] = React.useState([] as Rule[]);
+
+  const tableColumns = [
+    {
+      title: 'Name',
+      transforms: [ sortable ],
+    },
+    { title: 'Description', },
+    {
+      title: 'Match Expression',
+      transforms: [
+        info({
+          tooltip: 'A code-snippet expression which must evaluate to a boolean when applied to a given target. If the expression evaluates to true then the rule applies to that target.'
+        })
+      ],
+    },
+    {
+      title: 'Archival Period',
+      transforms: [
+        info({
+          tooltip: 'Period in seconds. Cryostat will connect to matching targets at this interval and copy the relevant recording data into its archives.'
+        })
+      ],
+    },
+    {
+      title: 'Preserved Archives',
+      transforms: [
+        info({
+          tooltip: 'The number of recording copies to be maintained in the Cryostat archives. Cryostat will continue retrieving further archived copies and trimming the oldest copies from the archive to maintain this limit.'
+        })
+      ],
+    },
+    {
+      title: 'Maximum Age',
+      transforms: [
+        info({
+          tooltip: 'The maximum age in seconds for data kept in the JFR recordings started by this rule.'
+        })
+      ],
+    },
+    {
+      title: 'Maximum Size',
+      transforms: [
+        info({
+          tooltip: 'The maximum size in bytes for JFR recordings started by this rule.'
+        })
+      ],
+    },
+  ] as ICell[];
+
+  const refreshRules = React.useCallback(() => {
+    setIsLoading(true);
+    addSubscription(
+      context.api.doGet('rules', 'v2').subscribe((v: any) => {
+        setRules(v.data.result);
+        setIsLoading(false);
+      })
+    );
+  }, [setIsLoading, addSubscription, context, context.api, setRules]);
+
+  React.useEffect(() => {
+    refreshRules();
+  }, [context, context.api]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.notificationChannel.messages(NotificationCategory.RuleCreated)
+        .subscribe(v => setRules(old => old.concat(v.message)))
+    );
+  }, [addSubscription, context, context.notificationChannel, setRules]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.notificationChannel.messages(NotificationCategory.RuleDeleted)
+        .subscribe(v => setRules(old => old.filter(o => o.name != v.message.name)))
+    )
+  }, [addSubscription, context, context.notificationChannel, setRules]);
+
+  React.useEffect(() => {
+    if (!context.settings.autoRefreshEnabled()) {
+      return;
+    }
+    const id = window.setInterval(() => refreshRules(), context.settings.autoRefreshPeriod() * context.settings.autoRefreshUnits());
+    return () => window.clearInterval(id);
+  }, []);
+
+  const handleSort = (event, index, direction) => {
```

Comment:
Could you explain how to sort the table? I tried adding a few rules with different names and clicking the arrow. The rules stayed in the same order.
![image](https://user-images.githubusercontent.com/84587295/154581267-eea5fa36-fb37-4b9e-86a3-31586d8b5960.png)